File: frontend/lib/data/model/data_responseDTO.dart
import 'responseDTO.dart';

class DataResponseDTO<T> extends ResponseDTO {
  final T? data;

  DataResponseDTO({
    required String status,
    required int code,
    required String message,
    this.data,
  }) : super(status: status, code: code, message: message);

  int getCode() {
    return code;
  }

  factory DataResponseDTO.fromJson(Map<String, dynamic> json,
      T Function(Map<String, dynamic>) fromDataJson) {
    return DataResponseDTO<T>(
      status: json['status'],
      code: json['code'],
      message: json['message'],
      data: json['data'] != null ? fromDataJson(json['data']) : null,
    );
  }
}

File: backend/src/main/java/com/aplus/aplusmarket/dto/DataResponseDTO.java
//package com.aplus.aplusmarket.dto;
//
//import lombok.*;
//import lombok.experimental.SuperBuilder;
//
///*
//    2024.1.26 하진희 :  responseDTO (data필요시)
// */
//@Getter
//@Setter
//@ToString
//@NoArgsConstructor
//public class DataResponseDTO<T> extends ResponseDTO {
//    private T data;
//
//    private DataResponseDTO(T data){
//        super();
//        this.data = data;
//    }
//
//    public DataResponseDTO(T data,int code,String message){
//        success()
//        this.data=data;
//    }
//
//
//    public DataResponseDTO(T data,int code){
//        this.data=data;
//    }
//
//
//    public static <T> DataResponseDTO<T> of(T data){
//        return new DataResponseDTO<T>(data);
//    }
//    public static <T> DataResponseDTO<T> of(T data,int code, String message){
//        return new DataResponseDTO<T>(data,code,message);
//    }
//
//    public static <T> DataResponseDTO<T> success(T data,int code){
//        return new DataResponseDTO<T>(data,code,"success");
//    }
//    public static <T> DataResponseDTO<T> empty(String message,int code){
//        return new DataResponseDTO<T>(null,code,message);
//    }
//}
